File: services/dispute/src/main/java/com/example/iter/dispute/dto/response/ReportSummaryResponse.java
package com.example.iter.dispute.dto.response;

import com.example.iter.auth.api.UserSummary;
import com.example.iter.dispute.domain.entity.ReportStatus;
import com.example.iter.dispute.domain.entity.ReportTargetType;

import java.time.LocalDateTime;

public record ReportSummaryResponse(
        Long reportId,
        UserSummary reporter,
        ReportTargetType targetType,
        Long targetId,
        String reason,
        ReportStatus status,
        LocalDateTime createdAt
) {
}

File: services/dispute/src/main/kotlin/iter/dispute/dto/response/ReportSummaryResponse.kt
package iter.dispute.dto.response

import iter.auth.api.UserSummary
import iter.dispute.domain.entity.ReportStatus
import iter.dispute.domain.entity.ReportTargetType
import java.time.LocalDateTime
import kotlin.jvm.JvmRecord

@JvmRecord
data class ReportSummaryResponse(
    val reportId: Long?,
    val reporter: UserSummary,
    val targetType: ReportTargetType,
    val targetId: Long,
    val reason: String,
    val status: ReportStatus,
    val createdAt: LocalDateTime?
)
